python/dolfinx/io/gmshio.py

```diff
@@ -33,6 +33,11 @@
 ]
 
 
+class TopologyDict(typing.TypedDict):
```

Comment:
What is this for? Should it be documented?

Reply:
It seems like `topologies` that that are defined on line 158 (and returned from `extract_topology_and_markers`) should be a dictionary with keys `topology` and `cell_data` with values being numpy arrays. In this case you can use a `TypedDict` to specify this for the type checker (see also https://peps.python.org/pep-0589/). In other words, the main difference from using a regular `dict[str, npt.NDArray[Any]]` is that you will get an error from `mypy` if you try so use a different key that what is specified in the `TypedDict`.

Reply:
A short docstring would help :)